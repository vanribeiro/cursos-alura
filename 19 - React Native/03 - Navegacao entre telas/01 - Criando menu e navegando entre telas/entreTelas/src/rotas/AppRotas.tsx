import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import ProdutorRotas from './ProdutorRotas';
import MelhoresProdutoresRotas from './MelhoresProdutoresRotas';
import Icon from 'react-native-vector-icons/Feather';

const Tab = createBottomTabNavigator();

function AppRotas() {
  const definirIcones =
    (route: any) =>
    // eslint-disable-next-line react/no-unstable-nested-components
    ({color}: any) => {
      return route.name === 'Melhores Produtores' ? (
        <Icon name="heart" size={22} color={color} />
      ) : (
        <Icon name="home" size={22} color={color} />
      );
    };

  return (
    <>
      <NavigationContainer>
        <Tab.Navigator
          screenOptions={({route}) => ({
            headerShown: false,
            tabBarStyle: {height: 60},
            tabBarLabelStyle: {
              fontSize: 12,
              paddingBottom: 4,
              fontFamily: 'Montserrat-Regular',
            },
            tabBarIcon: definirIcones(route),
            tabBarActiveTintColor: '#2a9f95',
            tabBarInactiveTintColor: '#c7c7c7',
          })}>
          <Tab.Screen name="Home" component={ProdutorRotas} />
          <Tab.Screen
            name="Melhores Produtores"
            component={MelhoresProdutoresRotas}
          />
        </Tab.Navigator>
      </NavigationContainer>
    </>
  );
}

export default AppRotas;
