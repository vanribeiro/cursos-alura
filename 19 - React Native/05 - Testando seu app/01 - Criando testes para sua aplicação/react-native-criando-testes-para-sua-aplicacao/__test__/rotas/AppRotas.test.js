import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import AppRotas from '../../src/rotas/AppRotas';
import { act } from 'react-test-renderer';

describe('rotas/AppRotas', () => {
    
    test('deve redenrizar os componentes de rotas', async () => {

        const { toJSON } = render(<AppRotas />);

        await waitFor(() => expect(toJSON()).toMatchSnapshot());
    });
});