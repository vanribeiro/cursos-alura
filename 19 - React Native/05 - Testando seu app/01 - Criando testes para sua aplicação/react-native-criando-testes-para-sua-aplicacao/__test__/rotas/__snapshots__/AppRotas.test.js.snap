// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rotas/AppRotas deve redenrizar os componentes de rotas 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Lista de LeilÃµes"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "paddingVertical": 8,
            }
          }
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onRefresh={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          refreshControl={
            <RefreshControlMock
              onRefresh={[Function]}
              refreshing={false}
            />
          }
          refreshing={false}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <RCTRefreshControl />
          <View />
        </RCTScrollView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
